components/playlist/playlist_media_file_downloader.cc

```diff
@@ -79,44 +89,141 @@ void PlaylistMediaFileDownloader::NotifySucceed(
   ResetDownloadStatus();
 }
 
+void PlaylistMediaFileDownloader::ScheduleToRemoveItem(
+    download::DownloadItem* item) {
+  for (auto& download : download_manager_->TakeInProgressDownloads()) {
+    DCHECK(download_item_observation_.IsObservingSource(download.get()));
+    in_progress_downloads_.push_back(std::move(download));
```

Comment:
and curious why we need `in_progress_downloads_` member?
Instead of scheduling item remove, how about calling `item->Remove()` here and
remove observation with `OnDownloadRemoved()`?
This could cause other problems?

Reply:
> Instead of scheduling item remove, how about calling item->Remove() here and

Yeah, that would be handy, but the problem is that we can't call `Remove()` inside download updated callback. So we should post task.

> curious why we need in_progress_downloads_ member?

Calling Remove() doesn't remove the item from the in_progress_downloads list. So we should take the list from the manager. And the download items should be alive until we call Remove().